examples/ner/run_ner.py

```diff
@@ -596,26 +600,69 @@ def main():
 
     args.model_type = args.model_type.lower()
     config_class, model_class, tokenizer_class = MODEL_CLASSES[args.model_type]
-    config = config_class.from_pretrained(
-        args.config_name if args.config_name else args.model_name_or_path,
-        num_labels=num_labels,
-        id2label={str(i): label for i, label in enumerate(labels)},
-        label2id={label: i for i, label in enumerate(labels)},
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
     tokenizer_args = {k: v for k, v in vars(args).items() if v is not None and k in TOKENIZER_ARGS}
     logger.info("Tokenizer arguments: %s", tokenizer_args)
     tokenizer = tokenizer_class.from_pretrained(
         args.tokenizer_name if args.tokenizer_name else args.model_name_or_path,
         cache_dir=args.cache_dir if args.cache_dir else None,
         **tokenizer_args,
     )
-    model = model_class.from_pretrained(
-        args.model_name_or_path,
-        from_tf=bool(".ckpt" in args.model_name_or_path),
-        config=config,
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
+
+    if args.has_new_labels and os.path.exists(args.model_name_or_path):
+        old_config = config_class.from_pretrained(
+            args.config_name if args.config_name else args.model_name_or_path,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_model = model_class.from_pretrained(
+            args.model_name_or_path,
+            from_tf=bool(".ckpt" in args.model_name_or_path),
+            config=old_config,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_num_labels = old_config.num_labels
+        new_labels = [label for label in labels if label not in old_config.label2id.keys()]
+        id2label = {**old_config.id2label, **{i+old_num_labels: label for i, label in enumerate(new_labels)}}
+        label2id = {**old_config.label2id, **{label: i+old_num_labels for i, label in enumerate(new_labels)}}
+        config = config_class.from_pretrained(
+            args.config_name if args.config_name else args.model_name_or_path,
+            num_labels=num_labels,
+            id2label=id2label,
+            label2id=label2id,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+
+        model = BertForTokenClassification(config)
+
+        params1 = old_model.named_parameters()
+        params2 = model.named_parameters()
+        dict_params1 = dict(params1)
+        dict_params2 = dict(params2)
+        for name1, param1 in dict_params1.items():
+            if name1 in dict_params2:
+                if name1.startswith('classifier'):
+                    dict_params2[name1].data[:old_num_labels].copy_(param1.data)
+                else:
+                    dict_params2[name1].data.copy_(param1.data)
```

Comment:
I think it would be nice to keep the names consistent.
```suggestion
        params_old = old_model.named_parameters()
        params_new = model.named_parameters()
        dict_params_old = dict(params_old)
        dict_params_new = dict(params_new)
        for name_old, param_old in dict_params_old.items():
            if name_new in dict_params_new:
                if name_new.startswith('classifier'):
                    dict_params_new[name_old].data[:old_num_labels].copy_(param_old.data)
                else:
                    dict_params_new[name_old].data.copy_(param_old.data)
```

Reply:
Also, it may be nice to check if all the params in the new model were visited.

Reply:
Thank you for naming suggestions here. I'll apply separately since I've needed to keep editing this.